src/vcpkg/visualstudio.cpp

```diff
@@ -199,6 +461,40 @@ namespace vcpkg::VisualStudio
         return instances;
     }
 
+    static std::vector<ToolVersion> get_windows_sdk_versions(const Filesystem& fs)
+    {
+        std::vector<ToolVersion> toolVersions;
+
+        const auto& program_files_32_bit = get_program_files_32_bit().value_or_exit(VCPKG_LINE_INFO);
+
+        // Instances from vswhere
+        const Path vswhere_exe = program_files_32_bit / "Microsoft Visual Studio" / "Installer" / "vswhere.exe";
```

Comment:
At the end of the day, we're going to call vcvars.bat. Instead of asking visual studio via vswhere, we should mimic the lower level logic actually used by vcvars (query registry keys).

This will be more accurate, faster, and let us check for >=8.1 instead of hardcoding a list of versions.

Reply:
That needs the admin permission to ready register, is that needed?

Reply:
Also please note that some users may not use admin permission to install Windows SDK, which caused the related info will not be added into registry.

Reply:
Done.